app/src/main/java/org/mozilla/fenix/settings/WallpaperSettingsFragment.kt

```diff
@@ -0,0 +1,175 @@
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.GridCells
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.fragment.app.Fragment
+import org.mozilla.fenix.R
+import org.mozilla.fenix.databinding.FragmentWallpaperSettingsBinding
+import org.mozilla.fenix.ext.requireComponents
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.wallpapers.Wallpaper
+import org.mozilla.fenix.wallpapers.WallpaperManager
+
+class WallpaperSettingsFragment : Fragment() {
+    private val binding by lazy {
+        FragmentWallpaperSettingsBinding.inflate(layoutInflater)
+    }
+
+    private val wallpaperManager by lazy {
+        requireComponents.wallpaperManager
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        binding.composeView.apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                FirefoxTheme {
+                    WallpaperSettings(wallpaperManager)
+                }
+            }
+        }
+        return binding.root
+    }
+}
+
+@Composable
+fun WallpaperSettings(
+    wallpaperManager: WallpaperManager
+) {
+    val currentlySelectedWallpaper = wallpaperManager.currentWallpaper.collectAsState()
+    Column {
+        Surface {
+            WallpaperThumbnails(
+                wallpapers = wallpaperManager.wallpapers.toList(),
+                onSelectionChanged = { wallpaperManager.updateWallpaperSelection(it) },
+                selectedWallpaper = currentlySelectedWallpaper.value
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun WallpaperThumbnails(
```

Comment:
I am somewhat shocked that ktlint didn't complain about how there's no kDoc for public functions, but I would definitely add some kDocs for all these public composable functions. Better get use to it early because I will ask for docs.

Reply:
🤦 you'd think that would be at the top of my brain, as a previous library author. I'll add some docs throughout

Reply:
Unrelated, but I'm also wondering what the best location for composables is. Should they be in a separate file? If so, do they belong under the `settings` package or the new `wallpaper` package?

Reply:
You can always refactor this as a follow up. Looking at our existing organization, I would imagine this would under `settings/wallpaper`. This is based on the idea that secondary(Wallpaper) fragments are packaged under their parent(Setting) fragment. While this isn't very formal by any means, I would try to observe the current patterns/structure and try to adapt. There's nothing wrong with going full maverick and making changes, but I would do that in a separate issue and ensure people are align with the changes we want to make for our Fenix architecture. I think the more important bits is that we want others to be able to adopt the styles.

Reply:
> I am somewhat shocked that ktlint didn't complain about how there's no kDoc for public functions

It's disabled in detekt here:
https://github.com/mozilla-mobile/fenix/blob/main/config/detekt.yml#L43

But enabled in A-C:
https://github.com/mozilla-mobile/android-components/blob/main/config/detekt.yml#L37

🤷‍♂️

Reply:
Filed https://github.com/mozilla-mobile/fenix/issues/23046 to investigate the difference between Fenix and AC, and hopefully align to something more common.